src/Futurelink.vue

```diff
@@ -10,62 +10,94 @@
-      // Don't run on the server
-      if (typeof window === 'undefined') {
-        return;
+    beforeCreate() {
+      // Ensure vue-router is actually being used (not just dependend on).
+      if (!this.$router || !this.$route) {
+        throw new Error('vue-futurelink requires vue-router to function.');
       }
-
-      // Don't run on mobile
-      if ('ontouchstart' in window) {
-        return;
+    },
+    data() {
+      return {
+        basePath: this.$router.options.base ? this.$router.options.base : '/',
+        preloaded: new Set(),
+        preloadComponent: undefined,
+        instance: undefined,
+        options: {
+          future: this.preloadLink,
+          links: [],
+        },
       }
+    },
+    destroyed() {
+      if (this.instance) {
+        this.instance.teardown();
+      }
+    },
+    methods: {
+      preloadLink(link) {
+        const href = link.getAttribute('href').replace(this.basePath, '/');
 
-      const loaded = [this.$route.path];
-
-      this.options.future = (link) => {
-        let href = link.getAttribute('href');
-
-        if (this.$router.options.base) {
-            href = href.replace(this.$router.options.base, '/')
-        }
-
-        if (loaded.includes(href)) {
+        // Immediately return if this route has already been preloaded.
+        if (this.preloaded.has(href)) {
           return;
         }
 
-        loaded.push(href);
-
-        this.$emit('preload', href);
+        // Route hasn't been preloaded, indicate that is has
+        this.preloaded.add(href);
 
         const resolved = this.$router.resolve(href);
         const matched = resolved.resolved.matched;
-        this.preloadComponent = matched[matched.length - 1].components.default;
-      };
+        const route = matched[matched.length - 1];
+        if (!route) {
+          return;
+        }
 
-      this.instance = futurelink(this.options);
+        // Create a promise to check if the route should be preloaded.
+        const shouldPreload = () => new Promise((resolve) => {
+          if (typeof route.meta.preload === 'function') {
+            try {
+              return resolve(route.meta.preload.call(this, href, route));
+            }
+                // Silence any errors; preloading is meant to be passive.
+            catch (e) {
+              return resolve(e);
+            }
+          }
+          resolve(route.meta.preload);
+        });
 
-      this.init();
-      this.$router.afterEach(() => this.init());
+        return shouldPreload()
+            .then((result) => {
+              // Do nothing if the result is an Error or explicitly set as false.
+              if (result instanceof Error || result === false) {
+                return;
+              }
+              this.$emit('preload', href, route);
+              this.preloadComponent = route.components.default;
+            })
+            // Silence any rejections; preloading is meant to be passive.
+            .catch(() => {});
```

Comment:
I agree that it shouldn't stop the execution of the script, but should it be completely silent? I think this would better with a `console.error()`.